parser_test.go

```diff
@@ -1257,6 +1258,98 @@ func TestArrayEach(t *testing.T) {
 	}, "a", "b")
 }
 
+func TestArrayEachNull(t *testing.T) {
+	count := 0
+	funcError := func([]byte, ValueType, int, error) {
+		count++
+	}
+
+	type args struct {
+		data []byte
+		cb   func(value []byte, dataType ValueType, offset int, err error)
+		keys []string
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantOffset int
+		wantErr    error
+		wantCount  int
+	}{
+		{
+			name: "Checking null in array, should be fine",
+			args: args{
+				data: []byte(`{"a":{"z":null,"b":[null],"c":null},"d":[]}`),
+				cb:   funcError,
+				keys: []string{"a", "b"},
+			},
+			wantOffset: 24,
+			wantErr:    nil,
+			wantCount:  1,
+		},
+		{
+			name: "Checking empty array, should be fine",
+			args: args{
+				data: []byte(`{"a":{"z":null,"b":[null],"c":null},"d":[]}`),
+				cb:   funcError,
+				keys: []string{"d"},
+			},
+			wantOffset: 41,
+			wantErr:    nil,
+			wantCount:  0,
+		},
+		{
+			name: "Checking null in c-key, should error ArrayEachNullError",
+			args: args{
+				data: []byte(`{"a":{"b":<HERE>["one","two"]},"c": null}`),
+				cb:   funcError,
+				keys: []string{"c"},
+			},
+			wantOffset: 36,
+			wantErr:    ArrayEachNullError,
+			wantCount:  0,
+		},
+		{
+			name: "Checking value in a, should error MalformedArrayError for missing subkey",
```

Comment:
Nit: The subkey isn't missing, rather it is an object instead of an array.

Reply:
Oh! Thanks, I'll make that correction.